### DISCUSSION TITLE
How to augment GraphQL with custom errors after a successful afterOperation

### DISCUSSION THREAD

**Discussion by ghost** [keystonejs/keystone#7469] [Questions] [answered]
Due to the nature of `afterOperation` hook and it being used to connect with other APIs / other actions, it could be really useful if there were an option to send a message back to the admin UI.

Maybe having a yellow prompt that confirms the database was updated, but there was an error with another integration.

**Comment by dcousens** [comment, marked as answer]
Hi @kennedybaird,

The GraphQL specification doesn't really support intermediate error messages as part of a successful operation.
We can't have meaningful `data` and `errors` intertwined without breaking that.

You could find multiple work-around solutions to this using custom mutations or manipulating purpose-built virtual fields,  but it would be a workaround and not well supported.
